third_party/xla_client/stablehlo_helper.cc

```diff
@@ -0,0 +1,44 @@
+#include "third_party/xla_client/stablehlo_helper.h"
+
+#include <iostream>
+
+#include "mlir/IR/Verifier.h"       // from @llvm-project
+#include "mlir/Pass/PassManager.h"  // from @llvm-project
+#include "tensorflow/compiler/xla/mlir_hlo/mhlo/transforms/passes.h"
+#include "tensorflow/compiler/xla/translate/hlo_to_mhlo/hlo_to_mlir_hlo.h"
+#include "tensorflow/compiler/xla/translate/mhlo_to_hlo/mlir_hlo_to_hlo.h"
+#include "third_party/xla_client/debug_macros.h"
+#include "third_party/xla_client/xla_util.h"
+
+namespace xla {
+
+static void hlo_mhlo_helper(const HloModuleProto* proto,
+                            mlir::ModuleOp* mlir_module) {
+  auto status = ConvertHloToMlirHlo(*mlir_module, proto,
+                                    /*import_all_computations*/ false);
+  XLA_CHECK(status.ok()) << "HLO -> MHLO conversion failed.";
+  XLA_CHECK(mlir::verify(*mlir_module).succeeded())
+      << "MHLO from HLO->MHLO is not leagal.";
```

Comment:
maybe add a line `Please open a github issue to PyTorch/XLA`, we should also dump the HLO graph that is illegal.

Reply:
Yes, that' s a good idea, let me add it

Reply:
Does the status message appear in logs anywhere as it is? Other wise consider including the `status.message()` as well.

Reply:
Yeah, that's a good idea to provide the error message to the user.